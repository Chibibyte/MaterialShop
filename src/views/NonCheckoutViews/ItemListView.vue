<template>
  <div>
    <v-container fluid class="pa-0 ma-0">
      <v-row justify="space-around" align="center" class="primary ma-0 pa-0">
        <v-col class="ma-0 pa-2" cols="4">
          <v-text-field
            dense
            href="searchbar"
            outlined
            class="ma-0 pa-0 white"
            placeholder="Name includes..."
            label="Name includes"
            v-model="searchValue"
            hide-details
          ></v-text-field>
        </v-col>
        <v-col class="ma-0 pa-0" cols="2">
          <v-select
            dense
            href="sortSelect"
            item-text="value"
            item-value="dir"
            :items="sortBy"
            hide-details
            class="ma-0 pa-0 white"
            outlined
            label="Sort by"
            v-model="sortValue"
          ></v-select>
        </v-col>
      </v-row>
    </v-container>

    <v-container class="white">
      <v-container class="d-flex">
        <v-pagination
          class="flex-grow-1 flex-shrink-0"
          v-model="page"
          @input="onChangePage"
          :length="maxPages"
          total-visible="10"
        ></v-pagination>
        <v-select
          v-model="maxItems"
          :items="maxItemsArr"
          class="flex-shrink-1 flex-grow-0"
          label="items"
          dense
          hide-details
          outlined
        ></v-select>
      </v-container>
      <v-row>
        <v-col
          v-for="item in items"
          :key="item.id"
          cols="12"
          xs="12"
          sm="6"
          md="4"
          lg="3"
          xl="2"
        >
          <ItemListViewItem :item="item" />
        </v-col>
      </v-row>
    </v-container>
  </div>
</template>

<script>
/**
 * Business Logic TODO:
 *
 * Inputs:
 *    -v-text-field:href:searchbar
 *    -v-select:href:sortSelect
 *    -$store.state.noncheckout.materialNames
 *    -$store.getters["noncheckout/items"]
 * Outputs:
 *    -v-text-field:href:searchbar
 *    -v-select:href:sortSelect
 */
import ItemListViewItem from "@/components/ItemListViewItem.vue";
export default {
  name: "ItemListView",
  components: {
    ItemListViewItem,
  },
  computed: {
    maxPages() {
      return Math.ceil(this.searchItems.length / this.maxItems);
    },
    searchItems() {
      let itemCpy = [...this.$store.getters["noncheckout/items"]];
      // search filter
      if (this.searchValue) {
        itemCpy = itemCpy.filter((item) =>
          item.title.includes(this.searchValue)
        );
      }
      return itemCpy;
    },
  },
  data() {
    return {
      page: 1,
      maxItemsArr: [20, 50, 100],
      maxItems: 0,
      items: [],
      sortValue: "Name ascending",
      sortBy: ["Name ascending", "Name descending"],
      searchValue: "",
    };
  },
  methods: {
    /**
     * Calculates the items to display based on the page @pageCount
     * "is user is on page @pageCount display @this.items"
     *
     * @param {Number} pageCount    Page to calculate items for
     */
    onChangePage(pageCount) {
      if (pageCount <= 0) return;
      let start = (pageCount - 1) * this.maxItems;
      let itemCpy = this.searchItems;
      // sort order
      if (this.sortValue === "Name descending") itemCpy.reverse();
      let items = itemCpy.slice(start, start + this.maxItems);
      this.items = items;
    },
  },
  watch: {
    maxItems() {
      this.onChangePage(this.page);
    },
    sortValue() {
      this.onChangePage(this.page);
    },
    searchValue() {
      this.page = 1;
      this.onChangePage(this.page);
    },
  },
  beforeMount() {
    this.maxItems = this.maxItemsArr[0];
    this.onChangePage(1);
  },
};
</script>
