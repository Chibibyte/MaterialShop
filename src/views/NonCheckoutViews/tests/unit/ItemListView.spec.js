
import Vuex from 'vuex'
import { shallowMount, createLocalVue } from '@vue/test-utils';
import ItemListView from '@/views/NonCheckoutViews/ItemListView.vue';

const localVue = createLocalVue();
localVue.use(Vuex);

let materialNames = [
    'alpha',
    'amount',
    'anna',
    'anno',
    'bread',
    'tennis',
    'toast'
]

let store;
let state;
let modules;
let stubs = [
    'v-container',
    'v-row',
    'v-col',
    'v-card',
    'v-card-title',
    'v-card-actions',
    'v-text-field',
    'v-pagination',
    'v-select'
]
let wrapper;

beforeEach(() => {
    state = {
        data() {
            return {
                maxItems: 0,
                searchValue: '',
                items: []
            }
        }
    }
    modules = {
        noncheckout: {
            namespaced: true,
            state() {
                return {
                    materialNames
                }
            },
            getters: {
                items: jest.fn((state) => {
                    let retVal = state.materialNames.map((name, index) => {
                        return { title: name, price: 2.99, id: index };
                    })
                    return retVal;
                })
            }
        }
    }
    store = new Vuex.Store({
        state,
        modules
    })
    wrapper = shallowMount(ItemListView, { stubs, store, localVue });
})

// initial
describe('Component', () => {
    test('is instantiated', () => {
        expect(wrapper.isVueInstance).toBeTruthy();
    });

    test('default snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    });
});

// methods
describe('methods', () => {
    function onChangePageTest(pageCount, searchValue, sortValue, maxItems, exp1, exp2) {
        test('onChangePage', async () => {
            wrapper.vm.items = [];
            modules.noncheckout.getters.items.mockClear();
            wrapper.vm.maxItems = maxItems;
            wrapper.vm.searchValue = searchValue;
            wrapper.vm.sortValue = sortValue;
            wrapper.vm.onChangePage(pageCount);
            expect(modules.noncheckout.getters.items.mock.calls).toHaveLength(exp1);
            expect(wrapper.vm.items).toHaveLength(exp2);
        })
    }

    // maxItems 10
    onChangePageTest(1, '', 'Name ascending', 10, 0, 7);
    onChangePageTest(1, '', 'Name descending', 10, 0, 7);
    onChangePageTest(1, 'an', 'Name ascending', 10, 0, 2);
    onChangePageTest(1, 'an', 'Name descending', 10, 0, 2);
    onChangePageTest(0, '', 'Name ascending', 10, 0, 0);
    onChangePageTest(0, '', 'Name descending', 10, 0, 0);


    // maxItems 1
    onChangePageTest(1, '', 'Name ascending', 1, 0, 1);
    onChangePageTest(1, '', 'Name descending', 1, 0, 1);
    onChangePageTest(1, 'an', 'Name ascending', 1, 0, 1);
    onChangePageTest(1, 'an', 'Name descending', 1, 0, 1);
    onChangePageTest(0, '', 'Name ascending', 1, 0, 0);
    onChangePageTest(0, '', 'Name descending', 1, 0, 0);

    // maxItems 3
    onChangePageTest(1, '', 'Name ascending', 3, 0, 3);
    onChangePageTest(1, '', 'Name descending', 3, 0, 3);
    onChangePageTest(1, 'an', 'Name ascending', 3, 0, 2);
    onChangePageTest(1, 'an', 'Name descending', 3, 0, 2);
    onChangePageTest(0, '', 'Name ascending', 3, 0, 0);
    onChangePageTest(0, '', 'Name descending', 3, 0, 0);

})

// watch
describe('watch', () => {
    function onChangePageTriggerTest(handlerName, value) {
        test(handlerName, async () => {
            wrapper.vm.onChangePage = jest.fn();
            await wrapper.vm.onChangePage.mockClear();
            await wrapper.setData({ [handlerName]: value })
            expect(wrapper.vm.onChangePage.mock.calls).toHaveLength(1);
            expect(wrapper.element).toMatchSnapshot();
        });
    }

    onChangePageTriggerTest('maxItems', 1);
    onChangePageTriggerTest('sortValue', 1);
    onChangePageTriggerTest('searchValue', 1);

    onChangePageTriggerTest('maxItems', 5);
    onChangePageTriggerTest('sortValue', 5);
    onChangePageTriggerTest('searchValue', 5);
})