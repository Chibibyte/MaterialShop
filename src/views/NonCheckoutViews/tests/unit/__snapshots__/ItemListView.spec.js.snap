// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component default snapshot 1`] = `
<div>
  <v-container-stub
    class="pa-0 ma-0"
    fluid=""
  >
    <v-row-stub
      align="center"
      class="primary ma-0 pa-0"
      justify="space-around"
    >
      <v-col-stub
        class="ma-0 pa-2"
        cols="4"
      >
        <v-text-field-stub
          class="ma-0 pa-0 white"
          dense=""
          hide-details=""
          href="searchbar"
          label="Name includes"
          outlined=""
          placeholder="Name includes..."
          value=""
        />
      </v-col-stub>
       
      <v-col-stub
        class="ma-0 pa-0"
        cols="2"
      >
        <v-select-stub
          class="ma-0 pa-0 white"
          dense=""
          hide-details=""
          href="sortSelect"
          item-text="value"
          item-value="dir"
          items="Name ascending,Name descending"
          label="Sort by"
          outlined=""
          value="Name ascending"
        />
      </v-col-stub>
    </v-row-stub>
  </v-container-stub>
   
  <v-container-stub
    class="white"
  >
    <v-container-stub
      class="d-flex"
    >
      <v-pagination-stub
        class="flex-grow-1 flex-shrink-0"
        length="1"
        total-visible="10"
        value="1"
      />
       
      <v-select-stub
        class="flex-shrink-1 flex-grow-0"
        dense=""
        hide-details=""
        items="20,50,100"
        label="items"
        outlined=""
        value="20"
      />
    </v-container-stub>
     
    <v-row-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
    </v-row-stub>
  </v-container-stub>
</div>
`;

exports[`watch maxItems 1`] = `
<div>
  <v-container-stub
    class="pa-0 ma-0"
    fluid=""
  >
    <v-row-stub
      align="center"
      class="primary ma-0 pa-0"
      justify="space-around"
    >
      <v-col-stub
        class="ma-0 pa-2"
        cols="4"
      >
        <v-text-field-stub
          class="ma-0 pa-0 white"
          dense=""
          hide-details=""
          href="searchbar"
          label="Name includes"
          outlined=""
          placeholder="Name includes..."
          value=""
        />
      </v-col-stub>
       
      <v-col-stub
        class="ma-0 pa-0"
        cols="2"
      >
        <v-select-stub
          class="ma-0 pa-0 white"
          dense=""
          hide-details=""
          href="sortSelect"
          item-text="value"
          item-value="dir"
          items="Name ascending,Name descending"
          label="Sort by"
          outlined=""
          value="Name ascending"
        />
      </v-col-stub>
    </v-row-stub>
  </v-container-stub>
   
  <v-container-stub
    class="white"
  >
    <v-container-stub
      class="d-flex"
    >
      <v-pagination-stub
        class="flex-grow-1 flex-shrink-0"
        length="8"
        total-visible="10"
        value="1"
      />
       
      <v-select-stub
        class="flex-shrink-1 flex-grow-0"
        dense=""
        hide-details=""
        items="20,50,100"
        label="items"
        outlined=""
        value="1"
      />
    </v-container-stub>
     
    <v-row-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
    </v-row-stub>
  </v-container-stub>
</div>
`;

exports[`watch maxItems 2`] = `
<div>
  <v-container-stub
    class="pa-0 ma-0"
    fluid=""
  >
    <v-row-stub
      align="center"
      class="primary ma-0 pa-0"
      justify="space-around"
    >
      <v-col-stub
        class="ma-0 pa-2"
        cols="4"
      >
        <v-text-field-stub
          class="ma-0 pa-0 white"
          dense=""
          hide-details=""
          href="searchbar"
          label="Name includes"
          outlined=""
          placeholder="Name includes..."
          value=""
        />
      </v-col-stub>
       
      <v-col-stub
        class="ma-0 pa-0"
        cols="2"
      >
        <v-select-stub
          class="ma-0 pa-0 white"
          dense=""
          hide-details=""
          href="sortSelect"
          item-text="value"
          item-value="dir"
          items="Name ascending,Name descending"
          label="Sort by"
          outlined=""
          value="Name ascending"
        />
      </v-col-stub>
    </v-row-stub>
  </v-container-stub>
   
  <v-container-stub
    class="white"
  >
    <v-container-stub
      class="d-flex"
    >
      <v-pagination-stub
        class="flex-grow-1 flex-shrink-0"
        length="2"
        total-visible="10"
        value="1"
      />
       
      <v-select-stub
        class="flex-shrink-1 flex-grow-0"
        dense=""
        hide-details=""
        items="20,50,100"
        label="items"
        outlined=""
        value="5"
      />
    </v-container-stub>
     
    <v-row-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
    </v-row-stub>
  </v-container-stub>
</div>
`;

exports[`watch searchValue 1`] = `
<div>
  <v-container-stub
    class="pa-0 ma-0"
    fluid=""
  >
    <v-row-stub
      align="center"
      class="primary ma-0 pa-0"
      justify="space-around"
    >
      <v-col-stub
        class="ma-0 pa-2"
        cols="4"
      >
        <v-text-field-stub
          class="ma-0 pa-0 white"
          dense=""
          hide-details=""
          href="searchbar"
          label="Name includes"
          outlined=""
          placeholder="Name includes..."
          value="1"
        />
      </v-col-stub>
       
      <v-col-stub
        class="ma-0 pa-0"
        cols="2"
      >
        <v-select-stub
          class="ma-0 pa-0 white"
          dense=""
          hide-details=""
          href="sortSelect"
          item-text="value"
          item-value="dir"
          items="Name ascending,Name descending"
          label="Sort by"
          outlined=""
          value="Name ascending"
        />
      </v-col-stub>
    </v-row-stub>
  </v-container-stub>
   
  <v-container-stub
    class="white"
  >
    <v-container-stub
      class="d-flex"
    >
      <v-pagination-stub
        class="flex-grow-1 flex-shrink-0"
        length="1"
        total-visible="10"
        value="1"
      />
       
      <v-select-stub
        class="flex-shrink-1 flex-grow-0"
        dense=""
        hide-details=""
        items="20,50,100"
        label="items"
        outlined=""
        value="20"
      />
    </v-container-stub>
     
    <v-row-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
    </v-row-stub>
  </v-container-stub>
</div>
`;

exports[`watch searchValue 2`] = `
<div>
  <v-container-stub
    class="pa-0 ma-0"
    fluid=""
  >
    <v-row-stub
      align="center"
      class="primary ma-0 pa-0"
      justify="space-around"
    >
      <v-col-stub
        class="ma-0 pa-2"
        cols="4"
      >
        <v-text-field-stub
          class="ma-0 pa-0 white"
          dense=""
          hide-details=""
          href="searchbar"
          label="Name includes"
          outlined=""
          placeholder="Name includes..."
          value="5"
        />
      </v-col-stub>
       
      <v-col-stub
        class="ma-0 pa-0"
        cols="2"
      >
        <v-select-stub
          class="ma-0 pa-0 white"
          dense=""
          hide-details=""
          href="sortSelect"
          item-text="value"
          item-value="dir"
          items="Name ascending,Name descending"
          label="Sort by"
          outlined=""
          value="Name ascending"
        />
      </v-col-stub>
    </v-row-stub>
  </v-container-stub>
   
  <v-container-stub
    class="white"
  >
    <v-container-stub
      class="d-flex"
    >
      <v-pagination-stub
        class="flex-grow-1 flex-shrink-0"
        length="1"
        total-visible="10"
        value="1"
      />
       
      <v-select-stub
        class="flex-shrink-1 flex-grow-0"
        dense=""
        hide-details=""
        items="20,50,100"
        label="items"
        outlined=""
        value="20"
      />
    </v-container-stub>
     
    <v-row-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
    </v-row-stub>
  </v-container-stub>
</div>
`;

exports[`watch sortValue 1`] = `
<div>
  <v-container-stub
    class="pa-0 ma-0"
    fluid=""
  >
    <v-row-stub
      align="center"
      class="primary ma-0 pa-0"
      justify="space-around"
    >
      <v-col-stub
        class="ma-0 pa-2"
        cols="4"
      >
        <v-text-field-stub
          class="ma-0 pa-0 white"
          dense=""
          hide-details=""
          href="searchbar"
          label="Name includes"
          outlined=""
          placeholder="Name includes..."
          value=""
        />
      </v-col-stub>
       
      <v-col-stub
        class="ma-0 pa-0"
        cols="2"
      >
        <v-select-stub
          class="ma-0 pa-0 white"
          dense=""
          hide-details=""
          href="sortSelect"
          item-text="value"
          item-value="dir"
          items="Name ascending,Name descending"
          label="Sort by"
          outlined=""
          value="1"
        />
      </v-col-stub>
    </v-row-stub>
  </v-container-stub>
   
  <v-container-stub
    class="white"
  >
    <v-container-stub
      class="d-flex"
    >
      <v-pagination-stub
        class="flex-grow-1 flex-shrink-0"
        length="1"
        total-visible="10"
        value="1"
      />
       
      <v-select-stub
        class="flex-shrink-1 flex-grow-0"
        dense=""
        hide-details=""
        items="20,50,100"
        label="items"
        outlined=""
        value="20"
      />
    </v-container-stub>
     
    <v-row-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
    </v-row-stub>
  </v-container-stub>
</div>
`;

exports[`watch sortValue 2`] = `
<div>
  <v-container-stub
    class="pa-0 ma-0"
    fluid=""
  >
    <v-row-stub
      align="center"
      class="primary ma-0 pa-0"
      justify="space-around"
    >
      <v-col-stub
        class="ma-0 pa-2"
        cols="4"
      >
        <v-text-field-stub
          class="ma-0 pa-0 white"
          dense=""
          hide-details=""
          href="searchbar"
          label="Name includes"
          outlined=""
          placeholder="Name includes..."
          value=""
        />
      </v-col-stub>
       
      <v-col-stub
        class="ma-0 pa-0"
        cols="2"
      >
        <v-select-stub
          class="ma-0 pa-0 white"
          dense=""
          hide-details=""
          href="sortSelect"
          item-text="value"
          item-value="dir"
          items="Name ascending,Name descending"
          label="Sort by"
          outlined=""
          value="5"
        />
      </v-col-stub>
    </v-row-stub>
  </v-container-stub>
   
  <v-container-stub
    class="white"
  >
    <v-container-stub
      class="d-flex"
    >
      <v-pagination-stub
        class="flex-grow-1 flex-shrink-0"
        length="1"
        total-visible="10"
        value="1"
      />
       
      <v-select-stub
        class="flex-shrink-1 flex-grow-0"
        dense=""
        hide-details=""
        items="20,50,100"
        label="items"
        outlined=""
        value="20"
      />
    </v-container-stub>
     
    <v-row-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
      <v-col-stub
        cols="2"
      >
        <itemlistviewitem-stub
          initamount="1"
          item="[object Object]"
        />
      </v-col-stub>
    </v-row-stub>
  </v-container-stub>
</div>
`;
